From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: factfinding <970308389@qq.com>
Date: Fri, 16 Oct 2026 15:00:39 +0800
Subject: [PATCH 6/7] v8: disable ETW stack walking for Windows 7

The api.cc hunk is from e3kskoy7wqk/Electron-for-windows-7@1086a0a v44.2.0/chromium/531c83c498a232a8f9f59d921437eced663754ad.diff. Its v8.gni hunk does not apply to Node's gyp build; the same switch is flipped in tools/v8_gypfiles/features.gypi instead.
---
 deps/v8/src/api/api.cc          | 4 ++++
 tools/v8_gypfiles/features.gypi | 3 ++-
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/deps/v8/src/api/api.cc b/deps/v8/src/api/api.cc
index 7dc060bc..19fc373d 100644
--- a/deps/v8/src/api/api.cc
+++ b/deps/v8/src/api/api.cc
@@ -11016,6 +11016,10 @@ void Isolate::SetFilterETWSessionByURL2Callback(
   i::Isolate* i_isolate = reinterpret_cast<i::Isolate*>(this);
   i_isolate->SetFilterETWSessionByURL2Callback(callback);
 }
+#else
+void Isolate::SetFilterETWSessionByURL2Callback(
+    FilterETWSessionByURL2Callback callback) {
+}
 #endif  // V8_ENABLE_ETW_STACK_WALKING
 
 bool v8::Object::IsCodeLike(v8::Isolate* v8_isolate) const {
diff --git a/tools/v8_gypfiles/features.gypi b/tools/v8_gypfiles/features.gypi
index 53886830..fd54544c 100644
--- a/tools/v8_gypfiles/features.gypi
+++ b/tools/v8_gypfiles/features.gypi
@@ -69,7 +69,8 @@
         'v8_enable_system_instrumentation': 0,
       }],
       ['OS == "win"', {
-        'v8_enable_etw_stack_walking': 1,
+        # Disabled for Windows 7, as in Electron-for-windows-7's v8.gni patch.
+        'v8_enable_etw_stack_walking': 0,
       }, {
         'v8_enable_etw_stack_walking': 0,
       }],
